src/ngraph/frontend/onnx_import/utils/common.cpp

```diff
@@ -28,22 +29,65 @@ namespace ngraph
     {
         namespace common
         {
-            const NodeVector& add_provenance_tags(const Node& onnx_node,
-                                                  const NodeVector& ng_node_vector)
+            namespace detail
             {
-                const std::string node_name =
-                    onnx_node.get_name().empty() ? "unnamed node" : onnx_node.get_name();
-                const std::string provenance_tag =
-                    "<ONNX " + onnx_node.op_type() + " (" + node_name + ")>";
+                static std::string concat_strings(
+                    const std::vector<std::reference_wrapper<const std::string>>& outputs)
+                {
+                    const auto concat_with_comma =
+                        [](const std::string& accumulator,
+                           std::reference_wrapper<const std::string> next_output) {
+                            return accumulator + ", " + next_output.get();
+                        };
+
+                    return std::accumulate(outputs.begin() + 1,
+                                           outputs.end(),
+                                           outputs.begin()->get(),
+                                           concat_with_comma);
+                }
+
+                /// \brief Builds a provenance tag for an ONNX operator
+                static std::string build_provenance_tag(const Node& onnx_node)
+                {
+                    const auto output_names = concat_strings(onnx_node.get_output_names());
+                    return std::string{"<ONNX " + onnx_node.op_type() + " (" +
+                                       onnx_node.get_name() + " -> " + output_names + ")>"};
```

Comment:
I would get rid of this space, it look strange in the ouput `( ->`
```suggestion
                                       onnx_node.get_name() + "-> " + output_names + ")>"};
```